dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tool.java

```diff
@@ -455,4 +456,11 @@ public String getDefaultTestCwlParameterFile() {
     public void setDefaultTestCwlParameterFile(String defaultTestCwlParameterFile) {
         getDefaultPaths().put(DescriptorLanguage.FileType.CWL_TEST_JSON, defaultTestCwlParameterFile);
     }
+
+    public List<String> calculateDescriptorType() {
+        Set<DescriptorLanguage.FileType> set = this.getWorkflowVersions().stream().flatMap(tag -> tag.getSourceFiles().stream()).map(SourceFile::getType).collect(
+                Collectors.toSet());
+        return Arrays.stream(DescriptorLanguage.values()).filter(lang -> set.contains(lang.getFileType()))
+                .map(lang -> lang.toString().toUpperCase()).distinct().collect(Collectors.toList());
```

Comment:
Didn't see this earlier.  Just return lang.toString() because it gets confusing for others to use it. 

"I have known DescriptorLanguages, do I uppercase it? do I lowercase it?"

So in general, the database column should either store DescriptorLanguage enum itself or the DescriptorLanguage enum toString(), not titlecase, lowercase, uppercase,

Reply:
.